### PR TITLE
Drop TxMeta column from TxHistory table

### PR DESCRIPTION
**Pull request by SirTyson**
# Description

Rebased on #4226.

This change drops `txmeta` from the `txhistory` SQL table when BucketListDB is enabled.

Unfortunately, SQLite does not support conditional column deletion, so I've added a new persistent state field indicating whether or not `txmeta` is supported. This allows us to check if the `txmeta` column exists before executing the drop SQL transaction. 

Addresses #4211.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: lib/catch.hpp
Comment (dmkozh) [inline lib/catch.hpp:1]: Was this formatted by accident?

File: lib/sqlite/sqlite3.h [hunk 1]
```diff
@@ -146,9 +146,9 @@ extern "C" {
 ** [sqlite3_libversion_number()], [sqlite3_sourceid()],
 ** [sqlite_version()] and [sqlite_source_id()].
 */
-#define SQLITE_VERSION        "3.41.2"
-#define SQLITE_VERSION_NUMBER 3041002
-#define SQLITE_SOURCE_ID      "2023-03-22 11:56:21 0d1fc92f94cb6b76bffe3ec34d69cffde2924203304e8ffc4155597af0c191da"
+#define SQLITE_VERSION        "3.45.1"
```
Comment (dmkozh) [inline lib/sqlite/sqlite3.h:149]: Can we make a separate PR with deps update and no other logic changes?

File: src/util/xdrquery/XDRQueryEval.cpp [hunk 2]
```diff
@@ -472,6 +472,12 @@ FieldList::getFieldNames() const
     return names;
 }
 
+inline std::string
+format_as(const NullField&)
```
Comment (dmkozh) [inline src/util/xdrquery/XDRQueryEval.cpp:476]: What is this? Doesn't seem like it's used anywhere.

File: lib/sqlite/sqlite3.h (hunk 1 above)
Comment (SirTyson) [reply lib/sqlite/sqlite3.h:149]: There is, as mentioned this is rebased on top off https://github.com/stellar/stellar-core/pull/4226. I think your comments refer to changes in that PR.

Comment (dmkozh) [reply lib/sqlite/sqlite3.h:149]: Sorry, missed the description somehow.

**Review by graydon — CHANGES_REQUESTED** [review]
I believe you can reflect on the table structure directly and check to see if it has a given column. For example in sqlite it's something like this:

```
SELECT EXISTS (
  SELECT 1
  FROM pragma_table_info('txhistory')
  WHERE name = 'txmeta'
);
```

and for postgres I think it's something like this:

```
SELECT EXISTS (
  SELECT 1
  FROM information_schema.columns 
  WHERE 
    table_name = 'txhistory' AND 
    column_name = 'txmeta'
);
```

**Comment by SirTyson** [comment]
> I believe you can reflect on the table structure directly and check to see if it has a given column. For example in sqlite it's something like this:
> 
> ```
> SELECT EXISTS (
>   SELECT 1
>   FROM pragma_table_info('txhistory')
>   WHERE name = 'txmeta'
> );
> ```
> 
> and for postgres I think it's something like this:
> 
> ```
> SELECT EXISTS (
>   SELECT 1
>   FROM information_schema.columns 
>   WHERE 
>     table_name = 'txhistory' AND 
>     column_name = 'txmeta'
> );
> ```

Done

File: src/main/ApplicationImpl.cpp [hunk 3]
```diff
@@ -709,6 +709,8 @@ ApplicationImpl::validateAndLogConfig()
     {
         if (mConfig.isUsingBucketListDB())
         {
+            // Tx meta column no longer supported in BucketListDB
+            mPersistentState->dropTxMetaIfExists(*mDatabase);
```
Comment (MonsieurNicolas) [inline src/main/ApplicationImpl.cpp:713]: I think you need to make this schema change in the same place than other schema changes so that people can run the `upgrade-db` (some people don't give the account that runs core sql access to update schema)

Comment (SirTyson) [reply src/main/ApplicationImpl.cpp]: Done

**Review by marta-lokhova — APPROVED** [review]
lgtm!

**Comment by marta-lokhova** [comment]
r+ 31dabd1e366e0c764a4ae87d51c883e2e771e173
